formula-android-tests/src/test/java/com/instacart/formula/FragmentFlowRenderViewTest.kt

```diff
@@ -184,18 +184,37 @@ class FragmentFlowRenderViewTest {
         assertThat(activeContracts()).containsExactly(TestContract()).inOrder()
     }
 
+    @Test fun `add multiple fragments with same fragment key`() {
+        scenario.moveToState(Lifecycle.State.RESUMED)
+
+        navigateToTaskDetail(id = 1)
+        navigateToTaskDetail(id = 2)
+        navigateToTaskDetail(id = 1)
+
+        assertFragmentViewIsCreated(TestContractWithId(1))
+        assertThat(activeContracts()).containsExactly(TestContract(), TestContractWithId(1), TestContractWithId(2), TestContractWithId(1)).inOrder()
+
+        navigateBack()
+        navigateBack()
+
+        assertFragmentViewIsCreated(TestContractWithId(1))
+        assertThat(activeContracts()).containsExactly(TestContract(), TestContractWithId(1)).inOrder()
+    }
+
     private fun navigateBack() {
         scenario.onActivity { it.onBackPressed() }
     }
 
-    private fun navigateToTaskDetail() {
-        val detail = TestContractWithId(1)
+    private fun navigateToTaskDetail(id: Int = 1) {
+        scenario.onActivity {
+            it.navigateTo(TestContractWithId(id))
+        }
+    }
+
+    private fun assertFragmentViewIsCreated(key: FragmentKey) {
```

Comment:
Adding a check to make sure that view is actually created (making sure that `getFeature` is called).